query_resolver.go

```diff
@@ -0,0 +1,103 @@
+package surrealdb
+
+import (
+	"context"
+	"time"
+
+	"github.com/buger/jsonparser"
+	"github.com/goccy/go-json"
+)
+
+type QueryResult[T any] struct {
+	Result []T    `json:"result"`
+	Status string `json:"status"`
+	Time   string `json:"time"`
+}
+
+type QueryParams = map[string]any
+
+type QueryResolver[T any] struct {
+	err          error
+	response     *RPCRawResponse
+	queryResults []QueryResult[T]
+	query        string
+	params       QueryParams
+	ctx          context.Context
+}
+
+// Query creates a new query resolver
+// It would be ideal if we create a global for *DB
+func Query[T any](db *DB, ctx context.Context, query string, params ...any) *QueryResolver[T] {
+	var paramsData = QueryParams{}
+	if len(params) > 0 {
+		paramsData = params[0].(QueryParams)
+	}
+
+	resolver := &QueryResolver[T]{
+		query:  query,
+		params: paramsData,
+		ctx:    ctx,
+	}
+
+	return resolver.resolve(db)
+}
+
+func (r *QueryResolver[T]) resolve(db *DB) *QueryResolver[T] {
```

Comment:
The first line of this method should probably be something along the lines of 
```go
if r.err != nil {
    return r
}
```
so we don't work on an invalid object